library/ssl_misc.h

```diff
@@ -276,6 +276,22 @@ uint32_t mbedtls_ssl_get_extension_mask(unsigned int extension_type);
 #endif /* MBEDTLS_CIPHER_MODE_CBC */
 #endif /* MBEDTLS_USE_PSA_CRYPTO*/
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO) && defined(PSA_WANT_ALG_CCM)
+#if defined(PSA_WANT_KEY_TYPE_AES)
+#define MBEDTLS_SSL_HAVE_AES_CCM
+#endif
+#if defined(PSA_WANT_KEY_TYPE_CAMELLIA)
+#define MBEDTLS_SSL_HAVE_CAMELLIA_CCM
+#endif
+#elif (!defined(MBEDTLS_USE_PSA_CRYPTO) && defined(MBEDTLS_CCM_C))
+#if defined(MBEDTLS_AES_C)
+#define MBEDTLS_SSL_HAVE_AES_CCM
+#endif
+#if defined(MBEDTLS_CAMELLIA_C)
+#define MBEDTLS_SSL_HAVE_CAMELLIA_CCM
+#endif
+#endif /* [!]MBEDTLS_USE_PSA_CRYPTO && [PSA_WANT_ALG_CCM/MBEDTLS_CCM_C] */
+
 #if defined(MBEDTLS_SSL_PROTO_TLS1_2)
 
 /* This macro determines whether CBC is supported. */
```

Comment:
Pre-exising: while at it, can you remove `DES_C` from the list? I should have been removed in 3.0 but apparently we missed it. (I checked, and I think that's the only occurrence. I guess we originally missed it because we grepped `library/ssl*.c` instead of `library/ssl*` - I just did the same mistake when checking for other occurrences and first didn't understand why it was not finding this one...)